bootstrap/util/configowner.go

```diff
@@ -76,6 +76,11 @@ func (co ConfigOwner) IsControlPlaneMachine() bool {
 	return ok
 }
 
+// IsMachinePool checks if an unstructured object is a MachinePool.
+func (co ConfigOwner) IsMachinePool() bool {
+	return co.GetKind() == "MachinePool"
```

Comment:
Didn't notice this before, but should we also check that the APIGroup matches? (not the version)

Reply:
Will do

Reply:
Actually, wouldn't the Group already be checked by GetConfigOwner below? I can add an additional check here but all the ConfigOwners should have group `clusterv1.GroupVersion.Group` so it would be a redundant check

Reply:
Sounds good, if we're already checking it below that should be enough